go/vt/vttablet/sandboxconn/sandboxconn.go

```diff
@@ -491,13 +497,49 @@ func (sbc *SandboxConn) Tablet() *topodatapb.Tablet {
 	return sbc.tablet
 }
 
-func (sbc *SandboxConn) getNextResult() *sqltypes.Result {
+func (sbc *SandboxConn) getNextResult(stmt sqlparser.Statement) *sqltypes.Result {
 	if len(sbc.results) != 0 {
 		r := sbc.results[0]
 		sbc.results = sbc.results[1:]
 		return r
 	}
-	return SingleRowResult
+	if stmt == nil {
+		// if we didn't get a valid query, we'll assume we need a SELECT
+		return SingleRowResult
+	}
+	switch stmt.(type) {
+	case *sqlparser.Select,
+		*sqlparser.Union,
+		*sqlparser.Show,
+		*sqlparser.Explain,
+		*sqlparser.OtherRead:
+		return SingleRowResult
+	case *sqlparser.Set,
+		sqlparser.DDLStatement,
+		*sqlparser.AlterVschema,
+		*sqlparser.Use,
+		*sqlparser.OtherAdmin,
+		*sqlparser.SetTransaction,
+		*sqlparser.Savepoint,
+		*sqlparser.SRollback,
+		*sqlparser.Release:
+		return &sqltypes.Result{}
+	}
+
+	// for everything else we fake a single row being affected
+	return &sqltypes.Result{RowsAffected: 1}
```

Comment:
only dml will have rows affected, can this logic be changed to have dml statements written out and otherwise return singlerow. What you think?